datafusion/sql/src/planner.rs

```diff
@@ -5509,6 +5599,99 @@ mod tests {
         assert!(logical_plan("SELECT \"1\"").is_err());
     }
 
+    #[test]
+    fn test_constant_expr_eq_join() {
+        let sql = "SELECT id, order_id \
+            FROM person \
+            INNER JOIN orders \
+            ON person.id = 10";
+
+        let expected = "Projection: person.id, orders.order_id\
+        \n  Filter: person.id = Int64(10)\
+        \n    CrossJoin:\
+        \n      TableScan: person\
+        \n      TableScan: orders";
+        quick_test(sql, expected);
+    }
+
+    #[test]
+    fn test_right_left_expr_eq_join() {
+        let sql = "SELECT id, order_id \
+            FROM person \
+            INNER JOIN orders \
+            ON orders.customer_id * 2 = person.id + 10";
+
+        let expected = "Projection: person.id, orders.order_id\
+        \n  Inner Join: person.id + Int64(10) = orders.customer_id * Int64(2)\
+        \n    Projection: person.id, person.first_name, person.last_name, person.age, person.state, person.salary, person.birth_date, person.😀, person.id + Int64(10)\
```

Comment:
👍

Reply:
Sorry, do not get chance to take a look at the PR.
Could you please add one more test case :

````
        let sql = "SELECT *
            FROM person \
            INNER JOIN orders \
            ON orders.customer_id * 2 = person.id + 10"

````
This is to verify the new added projection will not introduce additional columns in the final result and make sure those temp projected columns will be trimmed.

Reply:
I do not see you have logic to trim the temp projected columns.

Reply:
> I do not see you have logic to trim the temp projected columns.
Yes, for `select *`, the final result will contains the additional columns.

Do you means we need to add a new projection to trim temp projected columns?

Reply:
Yes, they should be removed.